### PR TITLE
allow users to fill in profile and save to database

### PR DESCRIPTION
**Pull request by jj7lin**
Added a profile form that takes in the following input from the user:
-username
-age
-user gender
-preferred gender for friends
-education/work
-interests
-bio
Also allows user to choose a file for profile pic and will display the selected file name

### CODE REVIEW

File: src/profile.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      this.setState({username: event.target.value});
+      this.setState({age: event.target.value});
+      this.setState({bio: event.target.value});
+      this.setState({gender: event.target.value});
+  }
+
+  render() {
+    return (
+      <form onUpdate={this.handleUpdate}>
+      <h1>My Profile</h1>
```
Comment (echanlee) [inline src/profile.js]: pls fix indentation

File: src/profile.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+      <input type="file" name="file" onChange={this.myChangeHandler}/>
```
Comment (echanlee) [inline src/profile.js]: uhhh what is this file for?

File: src/profile.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+      
+      <p>Username: </p>
+
+      <input type='text' name='username' onChange={this.myChangeHandler}/>
+
+      <p>Age:</p>
+      <input type='text' name='age' onChange={this.myChangeHandler}/>
```
Comment (echanlee) [inline src/profile.js]: you can make it of type number instead and then you don't have to check if it's a number later. also might be good to set a min/max. https://www.w3schools.com/tags/att_input_type_number.asp

File: src/profile.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+      <p>Your Gender:</p>
+      <select fieldValue = {this.state.gender} onChange ={this.myChangeHandler}>
+          <option fieldValue = "Female">Female</option>
+          <option fieldValue = "Male">Male</option>
```
Comment (echanlee) [inline src/profile.js]: should probably add "Other" to be ~inclusive

File: src/profile.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+      </select>
+
+      <p>Your Preferred Gender for friends:</p>
+      <select fieldValue = {this.state.gender} onChange ={this.myChangeHandler}>
+          <option fieldValue = "Female">Female</option>
+          <option fieldValue = "Male">Male</option>
```
Comment (echanlee) [inline src/profile.js]: add both/all option

File: src/profile.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+      <input type='text' name='age' onChange={this.myChangeHandler}/>
+
+      <p>Your Gender:</p>
+      <select fieldValue = {this.state.gender} onChange ={this.myChangeHandler}>
+          <option fieldValue = "Female">Female</option>
+          <option fieldValue = "Male">Male</option>
+      </select>
+
+      <p>Your Preferred Gender for friends:</p>
+      <select fieldValue = {this.state.gender} onChange ={this.myChangeHandler}>
```
Comment (echanlee) [inline src/profile.js]: this should have it's own state variable. right now it's using the same as gender

File: src/profile.js [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+          <option fieldValue = "Female">Female</option>
+          <option fieldValue = "Male">Male</option>
+      </select>
+
+      <p>Education/Work:</p>
+      <input type='text' name='education' onChange={this.myChangeHandler}/>
```
Comment (echanlee) [inline src/profile.js]: how come these ones don't have a state variable as their value

File: src/profile.js [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+
+      <p>Your interests</p>
+      <input type='text' name='interests'onChange={this.myChangeHandler}/>
+
+       <p>Bio:</p>
+      <input type='text'name='bio' onChange={this.myChangeHandler} />
+
+<button>Update!</button>
```
Comment (echanlee) [inline src/profile.js]: indentation

File: src/profile.js [hunk 9]
```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class MyForm extends React.Component {
```
Comment (echanlee) [inline src/profile.js]: maybe change the class name to something more descriptive

File: src/profile.js [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+        errMsg = <strong>infieldValid input, please enter a number</strong>;
+      }
+    }
+    this.setState({errormessage: errMsg});
+    this.setState({[nameField]: fieldVal});
+  }
+
+  handleUpdate = (event)=> {
+      event.preventDefault();
+      alert("Your profile has been updated!")
```
Comment (echanlee) [inline src/profile.js]: this is where you should put the API call later to attach it to python

File: src/profile.js [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+    };
+  }
+
+
+  myChangeHandler = (event) => {
+    console.log(event.target.files[0])
+    let nameField = event.target.name;
+    let fieldVal = event.target.fieldValue;
+    let errMsg = '';
+    if (nameField === "age") {
```
Comment (echanlee) [inline src/profile.js]: don't need this

**Review by echanlee — CHANGES_REQUESTED** [review]

File: src/profile.js (hunk 2 above)
Comment (jj7lin) [reply src/profile.js]: @echanlee profile pic

Comment (echanlee) [reply src/profile.js]: shouly probably store this somewhere too. also change the name to be more descriptive

File: src/profile.js (hunk 3 above)
Comment (jj7lin) [reply src/profile.js]: ok i set min to 18 and max to 100

File: src/profile.js (hunk 4 above)
Comment (jj7lin) [reply src/profile.js]: ok

File: src/profile.js (hunk 5 above)
Comment (jj7lin) [reply src/profile.js]: ok

File: src/profile.js (hunk 7 above)
Comment (jj7lin) [reply src/profile.js]: sorry ma'am i have forgotten

File: src/profile.js (hunk 8 above)
Comment (jj7lin) [reply src/profile.js]: ok

File: src/profile.js (hunk 9 above)
Comment (jj7lin) [reply src/profile.js]: alright

File: src/profile.js (hunk 10 above)
Comment (jj7lin) [reply src/profile.js]: thanks for the tip

File: src/profile.js (hunk 11 above)
Comment (jj7lin) [reply src/profile.js]: ok :'(

File: src/profile.js (hunk 1 above)
Comment (jj7lin) [reply src/profile.js]: ok

**Comment by jj7lin** [comment]
made some changes as per team's request

**Review by wclui — APPROVED** [review]

**Pull request merged**